disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/exporters/StoredBatchExporter.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.exporters;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public interface StoredBatchExporter {
+
+  /**
+   * Reads data from the disk and attempts to export it.
+   *
+   * @param timeout The amount of time to wait for the wrapped exporter to finish.
+   * @param unit The unit of the time provided.
+   * @return TRUE if there was data available and it was successfully exported within the timeout
+   *     provided. FALSE if either of those conditions didn't meet.
+   * @throws IOException If an unexpected error happens.
+   */
+  boolean exportStoredBatch(long timeout, TimeUnit unit) throws IOException;
```

Comment:
Seems like this method has roughly the same semantics as `CompletableResultCode {Signal}Exporter#flush()`.

Can we get rid of this method and instead have users use `flush()` to force an read from disk and export?

Reply:
I think it can work yeah. My only concern with doing so would be that the expectations of someone using this method won't probably match the reality of what actually happens, in the sense that, I believe the `flush()`method suggests that all the unexported signals will be exported right away. Though for the case of the disk exporter, it would mean that the next available batch of signals in the queue will be sent, but not all the batches at once. So I'm wondering if that could cause misunderstandings and probably invalid issues to be created in the repo because of it.